LaTeX/TeX.sublime-syntax

```diff
@@ -108,6 +154,55 @@ contexts:
       scope: constant.character.space.tex
     - include: escaped-character
 
+  tex-constants:
+    - match: (\\){{constants}}{{endcs}}
+      scope: constant.language.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      push:
+        - tex-constant-value
+        - tex-constant-assignment
+    - match: (\\){{macroconst}}{{endcs}}
+      scope: constant.language.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+
+  tex-constant-assignment:
+    - match: =
+      scope: keyword.operator.assignment.tex
+      pop: 1
+    - include: else-pop
+    - include: paragraph-pop
+
+  tex-constant-value:
+    - match: ((-?)\d+(\.)\d*)({{dimunits}}?)
+      scope: meta.number.float.decimal.tex
+      captures:
+        1: constant.numeric.value.tex
+        2: keyword.operator.arithmetic.tex
+        3: punctuation.separator.decimal.tex
+        4: constant.numeric.suffix.tex
+    - match: ((-?)\d+)({{dimunits}}?)
+      scope: meta.number.integer.decimal.tex
+      captures:
+        1: constant.numeric.value.tex
+        2: keyword.operator.arithmetic.tex
+        3: constant.numeric.suffix.tex
+    - match: (`)(\\)?({{charbycode}}|.)
+      scope: meta.number.integer.tex
+      captures:
+        1: keyword.operator.tex
+        2: punctuation.definition.backslash.tex
+        3: constant.character.tex
```

Comment:
I wonder if those char codes refer to escape sequences in other syntaxes. If so, we'd probably want to scope them `constant.character.escape` and drop `meta.number` as this has been used for number literals only.

```suggestion
    - match: (`)((\\)?(?:{{charbycode}}|.))
      captures:
        1: keyword.operator.tex
        2: constant.character.escape.tex
        3: punctuation.definition.backslash.tex
```

Reply:
The main purposes of the ^^ sequences, as I understand it, is to insert characters that you don't have on your keyboard, or are otherwise un-printable. So `^^Z`, for example, is not an escape sequence for the letter `Z`, but instead refers to, I think, the character in ascii code that you get if you subtract 65 from the character code of `Z`.

The `meta.number` is here, because the backtick is the operation "get the number associated with the character", so that the entire expression is just a convoluted way of getting a number.

Reply:
Sounds like something such as unicode escapes (e.g.:`\u2041`) in python to me.

https://github.com/sublimehq/Packages/blob/4c02b6a139c6944a7e1a7cbe585f912de99bdcf1/Python/Python.sublime-syntax#L1976-L1981

Reply:
in that case, yes, it is somewhat comparable. However, with the backtick, the entire expression still is treated like a number, e.g. 
```
\abovedisplayskip=`\^^Zmm
```
is a valid quantity meaning 25 millimeters.

Reply:
OK, than let's keep it as is.